processing/src/test/java/io/druid/query/filter/BoundDimFilterTest.java

```diff
@@ -68,10 +75,37 @@ public void testSerDesBoundFilter() throws IOException
   @Test
   public void testGetCacheKey()
   {
-    BoundDimFilter boundDimFilter = new BoundDimFilter("dimension", "12", "15", null, null, true);
-    BoundDimFilter boundDimFilterCopy = new BoundDimFilter("dimension", "12", "15", false, false, true);
+    BoundDimFilter boundDimFilter = new BoundDimFilter("dimension", "12", "15", null, null, true, null);
+    BoundDimFilter boundDimFilterCopy = new BoundDimFilter("dimension", "12", "15", false, false, true, null);
     Assert.assertArrayEquals(boundDimFilter.getCacheKey(), boundDimFilterCopy.getCacheKey());
-    BoundDimFilter anotherBoundDimFilter = new BoundDimFilter("dimension", "12", "15", true, null, false);
+    BoundDimFilter anotherBoundDimFilter = new BoundDimFilter("dimension", "12", "15", true, null, false, null);
     Assert.assertFalse(Arrays.equals(anotherBoundDimFilter.getCacheKey(), boundDimFilter.getCacheKey()));
+
+    BoundDimFilter boundDimFilterWithExtract = new BoundDimFilter("dimension", "12", "15", null, null, true, extractionFn);
+    BoundDimFilter boundDimFilterWithExtractCopy = new BoundDimFilter("dimension", "12", "15", false, false, true, extractionFn);
+    Assert.assertArrayEquals(boundDimFilterWithExtract.getCacheKey(), boundDimFilterWithExtractCopy.getCacheKey());
+
+    byte[] cacheKey = new byte[] {0x0A, 0x01, 0x00, 0x00, 0x01, (byte) 0xFF, 0x64, 0x69, 0x6D, 0x65, 0x6E, 0x73, 0x69,
+                                  0x6F, 0x6E, (byte) 0xFF, 0x31, 0x35, (byte) 0xFF, 0x31, 0x32, (byte) 0xFF};
+
+    byte[] cacheKeyWithExtract = new byte[] {0x0A, 0x01, 0x00, 0x00, 0x01, (byte) 0xFF, 0x64, 0x69, 0x6D, 0x65, 0x6E, 0x73, 0x69,
```

Comment:
This test is gonna be really brittle to future changes in how the cache keys are computed. It'd probably be just as good to assert that the cache keys are different for something with extractions and something without extractions.